src/components/Alerts/DeleteDialog.tsx

```diff
@@ -0,0 +1,76 @@
+"use client";
+
+import { Tenant } from "@/types/tenants";
+import Button from "../button/Button";
+import { AlertDialogDescription, AlertDialogTitle, AlertDialogContent } from "./AlertDialog";
+import { Action, Cancel } from "@radix-ui/react-alert-dialog";
+import { apiRoutes, getAPIURL } from "@/data/api";
+import { useRouter } from "next/navigation";
+import { ComponentPropsWithoutRef, ElementRef, forwardRef, useState } from "react";
+import { routes } from "@/routes";
+import { Input } from "../input/Input";
+
+type DeleteDialogProps = {
+	id: Tenant["id"];
+};
+
+const DeleteDialog = forwardRef<
+	ElementRef<typeof AlertDialogContent>,
+	ComponentPropsWithoutRef<typeof AlertDialogContent> & DeleteDialogProps
+>(({ id }, ref) => {
+	const [confirmText, setConfirmText] = useState("");
+	const router = useRouter();
+	async function deleteTenant() {
+		const res = await fetch(getAPIURL(apiRoutes.tenants.detail(id)), {
+			method: "DELETE",
+			credentials: "include",
+			cache: "no-store",
+			headers: {
+				"Content-Type": "application/json"
+			}
+		});
+		if (res.status === 401) router.push(routes.login);
+		router.refresh();
```

Comment:
I think we shouldn't handle fetch here. This should only include the component and no custom logic